site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package uk.ac.ox.ndph.mts.site_service.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import uk.ac.ox.ndph.mts.site_service.exception.InitialisationError;
+import uk.ac.ox.ndph.mts.site_service.exception.ValidationException;
+import uk.ac.ox.ndph.mts.site_service.model.Site;
+import uk.ac.ox.ndph.mts.site_service.repository.EntityStore;
+import uk.ac.ox.ndph.mts.site_service.validation.ModelEntityValidation;
+
+/**
+ * Implement an SiteServiceInterface interface.
+ * Validation of site based on the input configuration regex fields before
+ * sending the entity to store in the fhir repository.
+ */
+@Service
+public class SiteServiceImpl implements SiteService {
+
+    private EntityStore<Site> siteStore;
+    private final ModelEntityValidation<Site> entityValidation;
+    private final Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
+
+    /**
+     *
+     * @param siteStore Site store interface
+     * @param entityValidation Site validation interface 
+     */
+    @Autowired
+    public SiteServiceImpl(EntityStore<Site> siteStore,
+                           ModelEntityValidation<Site> entityValidation) {
+        if (siteStore == null) {
```

Comment:
do we need these validation checks as we check if the objects that are wired via the constructor are null - but Spring would complain if it couldnt find any beans??

Reply:
Taking forward as tech debt.